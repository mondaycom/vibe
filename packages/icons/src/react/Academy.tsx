/* eslint-disable */
/* tslint:disable */
import * as React from 'react';
export interface AcademyProps extends React.SVGAttributes<SVGElement> {
size?: string | number;
}
const Academy: React.FC<AcademyProps> = ({size, ...props}) => (
  <svg viewBox="0 0 20 20" fill="currentColor" width={ size || "20" } height={ size || "20" } {...props}>
    <path d="M9.63884 2.34268C9.86376 2.21911 10.1362 2.21911 10.3612 2.34268L18.3612 6.7382C18.601 6.86997 18.75 7.12188 18.75 7.39551C18.75 7.66915 18.601 7.92106 18.3612 8.05283L15.8731 9.41994L15.9082 13.8724C15.9082 14.1639 15.7858 14.4057 15.6772 14.5686C15.5615 14.7422 15.4107 14.9019 15.2486 15.0452C14.9231 15.3329 14.4827 15.6148 13.9708 15.8624C12.9445 16.3588 11.5413 16.76 9.98945 16.76C8.43763 16.76 7.03444 16.3588 6.00809 15.8624C5.49625 15.6148 5.05585 15.3329 4.73029 15.0452C4.56821 14.9019 4.41742 14.7422 4.30171 14.5686C4.19312 14.4057 4.07077 14.1639 4.07077 13.8724V9.38898L3.33509 8.98479C3.293 9.05241 3.25341 9.1217 3.21623 9.19266C2.91361 9.77034 2.75 10.4999 2.75 11.4108C2.75 11.4108 2.75 11.4108 2.75 11.4108L2.7501 13.5588C2.75012 13.973 2.41435 14.3088 2.00013 14.3088C1.58592 14.3088 1.25012 13.9731 1.2501 13.5589L1.25 11.4109C1.25 10.3249 1.44475 9.34178 1.88752 8.49659C1.92936 8.41672 1.97323 8.33844 2.01911 8.26177L1.63885 8.05284C1.39903 7.92108 1.25 7.66916 1.25 7.39553C1.25 7.12189 1.39902 6.86997 1.63884 6.7382L9.63884 2.34268ZM4.47158 7.89771L10.0003 10.9353L16.4425 7.39553L10 3.85575L5.37354 6.39771H9.91656C10.3308 6.39771 10.6666 6.7335 10.6666 7.14771C10.6666 7.56193 10.3308 7.89771 9.91656 7.89771H4.47158ZM5.57077 10.2131V13.7651C5.59922 13.8011 5.64767 13.8541 5.7236 13.9212C5.92115 14.0958 6.23767 14.3072 6.66119 14.512C7.50554 14.9204 8.68669 15.26 9.98945 15.26C11.2922 15.26 12.4734 14.9204 13.3177 14.512C13.7412 14.3072 14.0578 14.0958 14.2553 13.9212C14.3305 13.8548 14.3787 13.8022 14.4073 13.7662L14.3795 10.2406L10.3615 12.4484C10.1365 12.5719 9.86406 12.5719 9.63914 12.4484L5.57077 10.2131Z"
      fill="currentColor" fillRule="evenodd" clipRule="evenodd" />
  </svg>
);
Academy.displayName = 'Academy';
export default Academy;
/* tslint:enable */
/* eslint-enable */
