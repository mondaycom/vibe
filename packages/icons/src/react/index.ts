/* eslint-disable */
/* tslint:disable */
export {default as Academy} from './Academy';
export {default as Activity} from './Activity';
export {default as Add} from './Add';
export {default as AddNewDoc} from './AddNewDoc';
export {default as AddSmall} from './AddSmall';
export {default as AddToTeam} from './AddToTeam';
export {default as AddUpdate} from './AddUpdate';
export {default as Alert} from './Alert';
export {default as AlignCenter} from './AlignCenter';
export {default as AlignLeft} from './AlignLeft';
export {default as AlignRight} from './AlignRight';
export {default as Announcement} from './Announcement';
export {default as API} from './API';
export {default as Apps} from './Apps';
export {default as Archive} from './Archive';
export {default as Attach} from './Attach';
export {default as Baseline} from './Baseline';
export {default as Basic} from './Basic';
export {default as Battery} from './Battery';
export {default as BlockQuote} from './BlockQuote';
export {default as Board} from './Board';
export {default as BoardPrivate} from './BoardPrivate';
export {default as BoardShareable} from './BoardShareable';
export {default as BoardTemplate} from './BoardTemplate';
export {default as Bold} from './Bold';
export {default as Bolt} from './Bolt';
export {default as Bookmark} from './Bookmark';
export {default as Broadcast} from './Broadcast';
export {default as Broom} from './Broom';
export {default as Bug} from './Bug';
export {default as Bullet} from './Bullet';
export {default as Bullets} from './Bullets';
export {default as Bulllet} from './Bulllet';
export {default as Calendar} from './Calendar';
export {default as Chart} from './Chart';
export {default as Check} from './Check';
export {default as Checkbox} from './Checkbox';
export {default as CheckList} from './CheckList';
export {default as ClassicFolder} from './ClassicFolder';
export {default as Clear} from './Clear';
export {default as Clipboard} from './Clipboard';
export {default as Close} from './Close';
export {default as CloseMedium} from './CloseMedium';
export {default as CloseRound} from './CloseRound';
export {default as CloseSmall} from './CloseSmall';
export {default as Code} from './Code';
export {default as Collapse} from './Collapse';
export {default as CollapseHorizontally} from './CollapseHorizontally';
export {default as CollapseRound} from './CollapseRound';
export {default as Column} from './Column';
export {default as Comment} from './Comment';
export {default as Completed} from './Completed';
export {default as Connect} from './Connect';
export {default as ConnectedDoc} from './ConnectedDoc';
export {default as ContentDirectory} from './ContentDirectory';
export {default as ConvertToItem} from './ConvertToItem';
export {default as ConvertToSubitem} from './ConvertToSubitem';
export {default as Counter} from './Counter';
export {default as Country} from './Country';
export {default as CreditCard} from './CreditCard';
export {default as Custom} from './Custom';
export {default as Cut} from './Cut';
export {default as Dashboard} from './Dashboard';
export {default as DashboardPrivate} from './DashboardPrivate';
export {default as Deactivate} from './Deactivate';
export {default as Delete} from './Delete';
export {default as Dependency} from './Dependency';
export {default as Description} from './Description';
export {default as DisabledUser} from './DisabledUser';
export {default as Divider} from './Divider';
export {default as Doc} from './Doc';
export {default as DocPrivate} from './DocPrivate';
export {default as DocShareable} from './DocShareable';
export {default as DocTemplate} from './DocTemplate';
export {default as DoubleCheck} from './DoubleCheck';
export {default as Downgrade} from './Downgrade';
export {default as Download} from './Download';
export {default as Drag} from './Drag';
export {default as Dropdown} from './Dropdown';
export {default as DropdownChevronDown} from './DropdownChevronDown';
export {default as DropdownChevronLeft} from './DropdownChevronLeft';
export {default as DropdownChevronRight} from './DropdownChevronRight';
export {default as DropdownChevronUp} from './DropdownChevronUp';
export {default as DueDate} from './DueDate';
export {default as Duplicate} from './Duplicate';
export {default as Edit} from './Edit';
export {default as Education} from './Education';
export {default as Email} from './Email';
export {default as Embed} from './Embed';
export {default as Emoji} from './Emoji';
export {default as Enter} from './Enter';
export {default as Erase} from './Erase';
export {default as Event} from './Event';
export {default as Expand} from './Expand';
export {default as ExternalPage} from './ExternalPage';
export {default as Favorite} from './Favorite';
export {default as Feedback} from './Feedback';
export {default as File} from './File';
export {default as Files} from './Files';
export {default as Filter} from './Filter';
export {default as Folder} from './Folder';
export {default as Form} from './Form';
export {default as Formula} from './Formula';
export {default as Forum} from './Forum';
export {default as Forward} from './Forward';
export {default as Fullscreen} from './Fullscreen';
export {default as FullscreenClose} from './FullscreenClose';
export {default as Gallery} from './Gallery';
export {default as Gantt} from './Gantt';
export {default as Gif} from './Gif';
export {default as Globe} from './Globe';
export {default as Graph} from './Graph';
export {default as Group} from './Group';
export {default as Guest} from './Guest';
export {default as Health} from './Health';
export {default as Heart} from './Heart';
export {default as Help} from './Help';
export {default as Hide} from './Hide';
export {default as Highlight} from './Highlight';
export {default as HighlightColorBucket} from './HighlightColorBucket';
export {default as Home} from './Home';
export {default as Idea} from './Idea';
export {default as Image} from './Image';
export {default as Inbox} from './Inbox';
export {default as Info} from './Info';
export {default as Integrations} from './Integrations';
export {default as Invite} from './Invite';
export {default as IPRestrictions} from './IPRestrictions';
export {default as Italic} from './Italic';
export {default as Item} from './Item';
export {default as ItemDefaultValues} from './ItemDefaultValues';
export {default as ItemHeightDouble} from './ItemHeightDouble';
export {default as ItemHeightSingle} from './ItemHeightSingle';
export {default as ItemHeightTriple} from './ItemHeightTriple';
export {default as Key} from './Key';
export {default as Keyboard} from './Keyboard';
export {default as Labs} from './Labs';
export {default as Launch} from './Launch';
export {default as Layout} from './Layout';
export {default as LearnMore} from './LearnMore';
export {default as Lines} from './Lines';
export {default as Link} from './Link';
export {default as Location} from './Location';
export {default as Locked} from './Locked';
export {default as LogIn} from './LogIn';
export {default as LogOut} from './LogOut';
export {default as LongText} from './LongText';
export {default as Mention} from './Mention';
export {default as Menu} from './Menu';
export {default as Microphone} from './Microphone';
export {default as Minimize} from './Minimize';
export {default as MinusSmall} from './MinusSmall';
export {default as Mirror} from './Mirror';
export {default as Mobile} from './Mobile';
export {default as MondayDoc} from './MondayDoc';
export {default as MondayLogoOutline} from './MondayLogoOutline';
export {default as Moon} from './Moon';
export {default as MoreActions} from './MoreActions';
export {default as MoreBelow} from './MoreBelow';
export {default as MoreBelowFilled} from './MoreBelowFilled';
export {default as Move} from './Move';
export {default as MoveArrowDown} from './MoveArrowDown';
export {default as MoveArrowLeft} from './MoveArrowLeft';
export {default as MoveArrowLeftDouble} from './MoveArrowLeftDouble';
export {default as MoveArrowLeftNarrow} from './MoveArrowLeftNarrow';
export {default as MoveArrowRight} from './MoveArrowRight';
export {default as MoveArrowRightNarrow} from './MoveArrowRightNarrow';
export {default as MoveArrowUp} from './MoveArrowUp';
export {default as Mute} from './Mute';
export {default as MyWeek} from './MyWeek';
export {default as NavigationChevronDown} from './NavigationChevronDown';
export {default as NavigationChevronLeft} from './NavigationChevronLeft';
export {default as NavigationChevronRight} from './NavigationChevronRight';
export {default as NavigationChevronUp} from './NavigationChevronUp';
export {default as NavigationDoubleChevronLeft} from './NavigationDoubleChevronLeft';
export {default as NewTab} from './NewTab';
export {default as Night} from './Night';
export {default as NoColor} from './NoColor';
export {default as Note} from './Note';
export {default as Notifications} from './Notifications';
export {default as NotificationsMuted} from './NotificationsMuted';
export {default as Numbers} from './Numbers';
export {default as Offline} from './Offline';
export {default as Open} from './Open';
export {default as Page} from './Page';
export {default as Paste} from './Paste';
export {default as Pause} from './Pause';
export {default as PDF} from './PDF';
export {default as Person} from './Person';
export {default as PersonRound} from './PersonRound';
export {default as Pin} from './Pin';
export {default as PinFull} from './PinFull';
export {default as Placeholder} from './Placeholder';
export {default as Play} from './Play';
export {default as Print} from './Print';
export {default as Prompt} from './Prompt';
export {default as PushNotification} from './PushNotification';
export {default as Quote} from './Quote';
export {default as Radio} from './Radio';
export {default as Recurring} from './Recurring';
export {default as RecycleBin} from './RecycleBin';
export {default as Redo} from './Redo';
export {default as Remove} from './Remove';
export {default as Reply} from './Reply';
export {default as ReplyAll} from './ReplyAll';
export {default as Retry} from './Retry';
export {default as Robot} from './Robot';
export {default as Rotate} from './Rotate';
export {default as ScheduledEmail} from './ScheduledEmail';
export {default as Search} from './Search';
export {default as Security} from './Security';
export {default as Send} from './Send';
export {default as Settings} from './Settings';
export {default as SettingsKnobs} from './SettingsKnobs';
export {default as Share} from './Share';
export {default as ShortText} from './ShortText';
export {default as Show} from './Show';
export {default as Shredder} from './Shredder';
export {default as Shuffle} from './Shuffle';
export {default as Sort} from './Sort';
export {default as SortAscending} from './SortAscending';
export {default as SortDescending} from './SortDescending';
export {default as Sound} from './Sound';
export {default as Status} from './Status';
export {default as StrikethroughS} from './StrikethroughS';
export {default as StrikethroughT} from './StrikethroughT';
export {default as Subitems} from './Subitems';
export {default as Sun} from './Sun';
export {default as Switch} from './Switch';
export {default as Switcher} from './Switcher';
export {default as Table} from './Table';
export {default as Tags} from './Tags';
export {default as Team} from './Team';
export {default as Text} from './Text';
export {default as TextBig} from './TextBig';
export {default as Textcolor} from './Textcolor';
export {default as TextColorIndicator} from './TextColorIndicator';
export {default as TextCopy} from './TextCopy';
export {default as TextHuge} from './TextHuge';
export {default as TextMedium} from './TextMedium';
export {default as TextSmall} from './TextSmall';
export {default as ThumbsDown} from './ThumbsDown';
export {default as ThumbsUp} from './ThumbsUp';
export {default as Time} from './Time';
export {default as Timeline} from './Timeline';
export {default as Translation} from './Translation';
export {default as TurnInto} from './TurnInto';
export {default as Underline} from './Underline';
export {default as Undo} from './Undo';
export {default as Unlocked} from './Unlocked';
export {default as Update} from './Update';
export {default as Upgrade} from './Upgrade';
export {default as Upload} from './Upload';
export {default as UserDomain} from './UserDomain';
export {default as UserStatus} from './UserStatus';
export {default as Versioning} from './Versioning';
export {default as Video} from './Video';
export {default as Wand} from './Wand';
export {default as Warning} from './Warning';
export {default as WarningFull} from './WarningFull';
export {default as WhatsNew} from './WhatsNew';
export {default as Widgets} from './Widgets';
export {default as Work} from './Work';
export {default as Workflow} from './Workflow';
export {default as Workspace} from './Workspace';
export {default as WrapText} from './WrapText';
/* tslint:enable */
/* eslint-enable */
