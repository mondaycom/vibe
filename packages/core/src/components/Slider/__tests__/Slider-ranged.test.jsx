import { vi, describe, it, expect } from "vitest";
import React from "react";
import { act } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { renderSliderInRangeMode } from "./sliderTestUtils.jsx";

vi.useFakeTimers();

vi.mock("../../TextField/TextField", () => ({
  default: props => {
    return <div data-testid="mock-text-field-comp">{JSON.stringify(props)}</div>;
  }
}));

it("a. Ranges Slider: basic renderer", async () => {
  const { asFragment } = await renderSliderInRangeMode();
  expect(asFragment().firstChild).toMatchSnapshot();
});

describe("b. Ranges Slider Active/Inactive", () => {
  it("01. show Tooltip when hover Start Thumb", async () => {
    let result;
    await act(async () => {
      const { asFragment, elThumbStart } = await renderSliderInRangeMode();
      userEvent.hover(elThumbStart);
      vi.advanceTimersByTime(999);
      result = asFragment().firstChild;
    });
    expect(result).toMatchSnapshot();
  });

  it("02. show Tooltip when hover End Thumb", async () => {
    let result;
    await act(async () => {
      const { asFragment, elThumbEnd } = await renderSliderInRangeMode();
      userEvent.hover(elThumbEnd);
      vi.advanceTimersByTime(999);
      result = asFragment().firstChild;
    });
    expect(result).toMatchSnapshot();
  });

  it("03. hide Tooltip when unhover Thumb", async () => {
    let result;
    await act(async () => {
      const { asFragment, elThumbStart } = await renderSliderInRangeMode();
      userEvent.hover(elThumbStart);
      vi.advanceTimersByTime(999);
      userEvent.unhover(elThumbStart);
      vi.advanceTimersByTime(999);
      result = asFragment().firstChild;
    });
    expect(result).toMatchSnapshot();
  });

  it("04. activate Start Thumb by Tab key press", async () => {
    let result;
    await act(async () => {
      const { asFragment } = await renderSliderInRangeMode();
      userEvent.tab();
      vi.advanceTimersByTime(999);
      result = asFragment().firstChild;
    });
    expect(result).toMatchSnapshot();
  });

  it("05. activate End Thumb by Tab key press", async () => {
    let result;
    await act(async () => {
      const { asFragment } = await renderSliderInRangeMode();
      userEvent.tab();
      vi.advanceTimersByTime(999);
      userEvent.tab();
      vi.advanceTimersByTime(999);
      result = asFragment().firstChild;
    });
    expect(result).toMatchSnapshot();
  });

  it("06. de-activate Thumb by Tab key press, when End Thumb focused", async () => {
    let result;
    await act(async () => {
      const { asFragment, elThumbEnd } = await renderSliderInRangeMode();
      elThumbEnd.focus();
      userEvent.tab();
      vi.advanceTimersByTime(999);
      result = asFragment().firstChild;
    });
    expect(result).toMatchSnapshot();
  });
});

describe("c. Ranges Slider Key Events", () => {
  it("01. decrease value by Left/Down keys press", async () => {
    let result;
    await act(async () => {
      const { asFragment, elThumbStart } = await renderSliderInRangeMode({ showValue: true });
      elThumbStart.focus();
      userEvent.keyboard("{arrowleft}");
      userEvent.keyboard("{arrowdown}");
      result = asFragment().firstChild;
    });
    expect(result).toMatchSnapshot();
  });

  it("02. increase value by Right/Up keys press", async () => {
    let result;
    await act(async () => {
      const { asFragment, elThumbEnd } = await renderSliderInRangeMode({ showValue: true });
      elThumbEnd.focus();
      userEvent.keyboard("{arrowright}");
      userEvent.keyboard("{arrowup}");
      result = asFragment().firstChild;
    });
    expect(result).toMatchSnapshot();
  });

  it("03. decrease value (step 10%) by PageDown key press", async () => {
    let result;
    await act(async () => {
      const { asFragment, elThumbEnd } = await renderSliderInRangeMode({ showValue: true });
      elThumbEnd.focus();
      userEvent.keyboard("{pagedown}");
      result = asFragment().firstChild;
    });
    expect(result).toMatchSnapshot();
  });

  it("04. increase value (step 10%) by PageUp key pres", async () => {
    let result;
    await act(async () => {
      const { asFragment, elThumbStart } = await renderSliderInRangeMode({ showValue: true });
      elThumbStart.focus();
      userEvent.keyboard("{pageup}");
      userEvent.keyboard("{pageup}");
      result = asFragment().firstChild;
    });
    expect(result).toMatchSnapshot();
  });

  it("05. change value by keys press in Step and Label mode", async () => {
    let result;
    await act(async () => {
      const { asFragment, elThumbStart } = await renderSliderInRangeMode({
        indicateSelection: true,
        showValue: true,
        step: 10
      });
      elThumbStart.focus();
      await userEvent.keyboard("{arrowright}");
      await userEvent.keyboard("{arrowright}");
      result = asFragment().firstChild;
    });
    expect(result).toMatchSnapshot();
  });
});
