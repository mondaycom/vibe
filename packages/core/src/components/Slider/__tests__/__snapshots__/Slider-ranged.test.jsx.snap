// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`a. Ranges Slider: basic renderer 1`] = `
<div
  class="slider positionTop"
  data-testid="monday-slider"
>
  <div
    class="base small"
    data-testid="monday-slider__base"
  >
    <div
      class="rail small"
      data-testid="monday-slider__rail"
    >
      <div
        class="track"
      />
      <div
        class="filledTrack"
        style="left: 25%; width: 40%;"
      />
      <div
        aria-disabled="false"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="25"
        aria-valuetext="25%"
        class="thumb small notDisabledThumb"
        data-testid="monday-slider__thumb-0"
        role="slider"
        style="left: 25%;"
        tabindex="0"
      />
      <div
        aria-disabled="false"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="65"
        aria-valuetext="65%"
        class="thumb small notDisabledThumb"
        data-testid="monday-slider__thumb-1"
        role="slider"
        style="left: 65%;"
        tabindex="0"
      />
    </div>
  </div>
</div>
`;

exports[`b. Ranges Slider Active/Inactive > 01. show Tooltip when hover Start Thumb 1`] = `
<div
  class="slider positionTop"
  data-testid="monday-slider"
>
  <div
    class="base small"
    data-testid="monday-slider__base"
  >
    <div
      class="rail small"
      data-testid="monday-slider__rail"
    >
      <div
        class="track"
      />
      <div
        class="filledTrack"
        style="left: 25%; width: 40%;"
      />
      <div
        aria-disabled="false"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="25"
        aria-valuetext="25%"
        class="thumb small notDisabledThumb"
        data-testid="monday-slider__thumb-0"
        role="slider"
        style="left: 25%;"
        tabindex="0"
      />
      <div
        aria-disabled="false"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="65"
        aria-valuetext="65%"
        class="thumb small notDisabledThumb"
        data-testid="monday-slider__thumb-1"
        role="slider"
        style="left: 65%;"
        tabindex="0"
      />
    </div>
  </div>
</div>
`;

exports[`b. Ranges Slider Active/Inactive > 02. show Tooltip when hover End Thumb 1`] = `
<div
  class="slider positionTop"
  data-testid="monday-slider"
>
  <div
    class="base small"
    data-testid="monday-slider__base"
  >
    <div
      class="rail small"
      data-testid="monday-slider__rail"
    >
      <div
        class="track"
      />
      <div
        class="filledTrack"
        style="left: 25%; width: 40%;"
      />
      <div
        aria-disabled="false"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="25"
        aria-valuetext="25%"
        class="thumb small notDisabledThumb"
        data-testid="monday-slider__thumb-0"
        role="slider"
        style="left: 25%;"
        tabindex="0"
      />
      <div
        aria-disabled="false"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="65"
        aria-valuetext="65%"
        class="thumb small notDisabledThumb"
        data-testid="monday-slider__thumb-1"
        role="slider"
        style="left: 65%;"
        tabindex="0"
      />
    </div>
  </div>
</div>
`;

exports[`b. Ranges Slider Active/Inactive > 03. hide Tooltip when unhover Thumb 1`] = `
<div
  class="slider positionTop"
  data-testid="monday-slider"
>
  <div
    class="base small"
    data-testid="monday-slider__base"
  >
    <div
      class="rail small"
      data-testid="monday-slider__rail"
    >
      <div
        class="track"
      />
      <div
        class="filledTrack"
        style="left: 25%; width: 40%;"
      />
      <div
        aria-disabled="false"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="25"
        aria-valuetext="25%"
        class="thumb small notDisabledThumb"
        data-testid="monday-slider__thumb-0"
        role="slider"
        style="left: 25%;"
        tabindex="0"
      />
      <div
        aria-disabled="false"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="65"
        aria-valuetext="65%"
        class="thumb small notDisabledThumb"
        data-testid="monday-slider__thumb-1"
        role="slider"
        style="left: 65%;"
        tabindex="0"
      />
    </div>
  </div>
</div>
`;

exports[`b. Ranges Slider Active/Inactive > 04. activate Start Thumb by Tab key press 1`] = `
<div
  class="slider positionTop"
  data-testid="monday-slider"
>
  <div
    class="base small"
    data-testid="monday-slider__base"
  >
    <div
      class="rail small"
      data-testid="monday-slider__rail"
    >
      <div
        class="track"
      />
      <div
        class="filledTrack"
        style="left: 25%; width: 40%;"
      />
      <div
        aria-disabled="false"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="25"
        aria-valuetext="25%"
        class="thumb small notDisabledThumb"
        data-testid="monday-slider__thumb-0"
        role="slider"
        style="left: 25%;"
        tabindex="0"
      />
      <div
        aria-disabled="false"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="65"
        aria-valuetext="65%"
        class="thumb small notDisabledThumb"
        data-testid="monday-slider__thumb-1"
        role="slider"
        style="left: 65%;"
        tabindex="0"
      />
    </div>
  </div>
</div>
`;

exports[`b. Ranges Slider Active/Inactive > 05. activate End Thumb by Tab key press 1`] = `
<div
  class="slider positionTop"
  data-testid="monday-slider"
>
  <div
    class="base small"
    data-testid="monday-slider__base"
  >
    <div
      class="rail small"
      data-testid="monday-slider__rail"
    >
      <div
        class="track"
      />
      <div
        class="filledTrack"
        style="left: 25%; width: 40%;"
      />
      <div
        aria-disabled="false"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="25"
        aria-valuetext="25%"
        class="thumb small focused notDisabledThumb"
        data-testid="monday-slider__thumb-0"
        role="slider"
        style="left: 25%;"
        tabindex="0"
      />
      <div
        aria-disabled="false"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="65"
        aria-valuetext="65%"
        class="thumb small focused notDisabledThumb"
        data-testid="monday-slider__thumb-1"
        role="slider"
        style="left: 65%;"
        tabindex="0"
      />
    </div>
  </div>
</div>
`;

exports[`b. Ranges Slider Active/Inactive > 06. de-activate Thumb by Tab key press, when End Thumb focused 1`] = `
<div
  class="slider positionTop"
  data-testid="monday-slider"
>
  <div
    class="base small"
    data-testid="monday-slider__base"
  >
    <div
      class="rail small"
      data-testid="monday-slider__rail"
    >
      <div
        class="track"
      />
      <div
        class="filledTrack"
        style="left: 25%; width: 40%;"
      />
      <div
        aria-disabled="false"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="25"
        aria-valuetext="25%"
        class="thumb small notDisabledThumb"
        data-testid="monday-slider__thumb-0"
        role="slider"
        style="left: 25%;"
        tabindex="0"
      />
      <div
        aria-disabled="false"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="65"
        aria-valuetext="65%"
        class="thumb small notDisabledThumb"
        data-testid="monday-slider__thumb-1"
        role="slider"
        style="left: 65%;"
        tabindex="0"
      />
    </div>
  </div>
</div>
`;

exports[`c. Ranges Slider Key Events > 01. decrease value by Left/Down keys press 1`] = `
<div
  class="slider valueShown positionTop"
  data-testid="monday-slider"
>
  <div
    class="base small"
    data-testid="monday-slider__base"
  >
    <div
      class="rail small"
      data-testid="monday-slider__rail"
    >
      <div
        class="track"
      />
      <div
        class="filledTrack"
        style="left: 23%; width: 42%;"
      />
      <div
        aria-disabled="false"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="23"
        aria-valuetext="23%"
        class="thumb small notDisabledThumb"
        data-testid="monday-slider__thumb-0"
        role="slider"
        style="left: 23%;"
        tabindex="0"
      >
        <label
          class="label colorPrimary positionTop"
        >
          23%
        </label>
      </div>
      <div
        aria-disabled="false"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="65"
        aria-valuetext="65%"
        class="thumb small notDisabledThumb"
        data-testid="monday-slider__thumb-1"
        role="slider"
        style="left: 65%;"
        tabindex="0"
      >
        <label
          class="label colorPrimary positionTop"
        >
          65%
        </label>
      </div>
    </div>
  </div>
</div>
`;

exports[`c. Ranges Slider Key Events > 02. increase value by Right/Up keys press 1`] = `
<div
  class="slider valueShown positionTop"
  data-testid="monday-slider"
>
  <div
    class="base small"
    data-testid="monday-slider__base"
  >
    <div
      class="rail small"
      data-testid="monday-slider__rail"
    >
      <div
        class="track"
      />
      <div
        class="filledTrack"
        style="left: 25%; width: 42%;"
      />
      <div
        aria-disabled="false"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="25"
        aria-valuetext="25%"
        class="thumb small focused notDisabledThumb"
        data-testid="monday-slider__thumb-0"
        role="slider"
        style="left: 25%;"
        tabindex="0"
      >
        <label
          class="label colorPrimary positionTop"
        >
          25%
        </label>
      </div>
      <div
        aria-disabled="false"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="67"
        aria-valuetext="67%"
        class="thumb small focused notDisabledThumb"
        data-testid="monday-slider__thumb-1"
        role="slider"
        style="left: 67%;"
        tabindex="0"
      >
        <label
          class="label colorPrimary positionTop"
        >
          67%
        </label>
      </div>
    </div>
  </div>
</div>
`;

exports[`c. Ranges Slider Key Events > 03. decrease value (step 10%) by PageDown key press 1`] = `
<div
  class="slider valueShown positionTop"
  data-testid="monday-slider"
>
  <div
    class="base small"
    data-testid="monday-slider__base"
  >
    <div
      class="rail small"
      data-testid="monday-slider__rail"
    >
      <div
        class="track"
      />
      <div
        class="filledTrack"
        style="left: 25%; width: 30%;"
      />
      <div
        aria-disabled="false"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="25"
        aria-valuetext="25%"
        class="thumb small focused notDisabledThumb"
        data-testid="monday-slider__thumb-0"
        role="slider"
        style="left: 25%;"
        tabindex="0"
      >
        <label
          class="label colorPrimary positionTop"
        >
          25%
        </label>
      </div>
      <div
        aria-disabled="false"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="55"
        aria-valuetext="55%"
        class="thumb small focused notDisabledThumb"
        data-testid="monday-slider__thumb-1"
        role="slider"
        style="left: 55%;"
        tabindex="0"
      >
        <label
          class="label colorPrimary positionTop"
        >
          55%
        </label>
      </div>
    </div>
  </div>
</div>
`;

exports[`c. Ranges Slider Key Events > 04. increase value (step 10%) by PageUp key pres 1`] = `
<div
  class="slider valueShown positionTop"
  data-testid="monday-slider"
>
  <div
    class="base small"
    data-testid="monday-slider__base"
  >
    <div
      class="rail small"
      data-testid="monday-slider__rail"
    >
      <div
        class="track"
      />
      <div
        class="filledTrack"
        style="left: 45%; width: 20%;"
      />
      <div
        aria-disabled="false"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="45"
        aria-valuetext="45%"
        class="thumb small notDisabledThumb"
        data-testid="monday-slider__thumb-0"
        role="slider"
        style="left: 45%;"
        tabindex="0"
      >
        <label
          class="label colorPrimary positionTop"
        >
          45%
        </label>
      </div>
      <div
        aria-disabled="false"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="65"
        aria-valuetext="65%"
        class="thumb small notDisabledThumb"
        data-testid="monday-slider__thumb-1"
        role="slider"
        style="left: 65%;"
        tabindex="0"
      >
        <label
          class="label colorPrimary positionTop"
        >
          65%
        </label>
      </div>
    </div>
  </div>
</div>
`;

exports[`c. Ranges Slider Key Events > 05. change value by keys press in Step and Label mode 1`] = `
<div
  class="slider valueShown positionTop"
  data-testid="monday-slider"
>
  <div
    class="infix prefix"
    style="width: 60px;"
  >
    <div
      data-testid="mock-text-field-comp"
    >
      {"value":"45%","className":"selectionIndicator"}
    </div>
  </div>
  <div
    class="base small"
    data-testid="monday-slider__base"
  >
    <div
      class="rail small"
      data-testid="monday-slider__rail"
    >
      <div
        class="track"
      />
      <div
        class="filledTrack"
        style="left: 45%; width: 20%;"
      />
      <div
        aria-disabled="false"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="45"
        aria-valuetext="45%"
        class="thumb small notDisabledThumb"
        data-testid="monday-slider__thumb-0"
        role="slider"
        style="left: 45%;"
        tabindex="0"
      >
        <label
          class="label colorPrimary positionTop"
        >
          45%
        </label>
      </div>
      <div
        aria-disabled="false"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="65"
        aria-valuetext="65%"
        class="thumb small notDisabledThumb"
        data-testid="monday-slider__thumb-1"
        role="slider"
        style="left: 65%;"
        tabindex="0"
      >
        <label
          class="label colorPrimary positionTop"
        >
          65%
        </label>
      </div>
    </div>
  </div>
  <div
    class="infix postfix"
    style="width: 60px;"
  >
    <div
      data-testid="mock-text-field-comp"
    >
      {"value":"65%","className":"selectionIndicator"}
    </div>
  </div>
</div>
`;
